docs/content/guides/dagster/enriching-with-software-defined-assets.mdx

```diff
@@ -0,0 +1,312 @@
+# Software-Defined Assets for Existing Dagster Users
```

Comment:
This doesn't explicitly apply to existing Dagster users, so to me, this title doesn't accurately convey what this guide is actually doing.

Reply:
Mind elaborating a little bit on what you mean by this?

Reply:
Sure. This guide, to me, is more than about existing vs new users - it's about the difference between SDA and ops/graphs/etc. This guide lays out when SDA is a better fit over other approaches, which is useful to both new *and* existing users.

Existing users might want to enrich their jobs (as you say in here!) and new users might not be sure what to do and are looking for guidance.

My concern is that the title calling out `existing users` may make new users feel like it's not applicable to them and then they miss out on some really good info.

I'm also not sure that this title will have the guide show up in a good spot in search results. The question for that is what would someone search for if they were trying to do the very thing this guide describes?

Reply:
Got it - those points make sense to me, but I'm struggling a little bit to think of a different name that I like better. Do you have one in mind?

One of the things that makes it "existing user"-y is that it assumes a lot of knowledge about existing Dagster concepts. It gives some basic definitions of ops and graphs, but it doesn't go out of its way to explain how they work.

Reply:
I just pushed up my changes and went with `Upgrading to Software-Defined Assets` to just have something, but I'm open to other suggestions.

`Upgrade` implies that something already exists, so perhaps that could satisfy the existing user part of this? TBH I don't know that a piece of documentation *needs* to explain in full all the concepts it touches on to be new user-friendly.  As long as we link out and give context for what the current piece focuses on, I think that's enough, at least for this!